# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1406903559 -7200
# Node ID 27ac00137d1a7a2cc2af1c0fa8b8198768568000
# Parent  42066a68f616cb2ae57c5dc384a140d7d9dc0e76
x86/cpu: drop the num_siblings check against nr_cpu_ids

The printk() is missing a newline which resulted in console corruption.

However, nr_cpu_ids can be legitimately lower than valid num_sibling values
given certain compile or boot time configuration.

Suggested-by: Jan Beulich <jbeulich@suse.com>
Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>

diff -r 42066a68f616 -r 27ac00137d1a xen/arch/x86/cpu/common.c
--- a/xen/arch/x86/cpu/common.c
+++ b/xen/arch/x86/cpu/common.c
@@ -443,13 +443,6 @@ void __cpuinit detect_ht(struct cpuinfo_
 	if (c->x86_num_siblings == 1) {
 		printk(KERN_INFO  "CPU: Hyper-Threading is disabled\n");
 	} else if (c->x86_num_siblings > 1 ) {
-
-		if (c->x86_num_siblings > nr_cpu_ids) {
-			printk(KERN_WARNING "CPU: Unsupported number of the siblings %d", c->x86_num_siblings);
-			c->x86_num_siblings = 1;
-			return;
-		}
-
 		index_msb = get_count_order(c->x86_num_siblings);
 		c->phys_proc_id = phys_pkg_id((ebx >> 24) & 0xFF, index_msb);
 
